src/coldcard.test.js

```diff
@@ -0,0 +1,609 @@
+
+  describe("request", () => {
+    it("return psbt if there is one", () => {
+      const interaction = interactionBuilder({
+        network: TESTNET,
+        psbt: singleInputB64PSBT,
+      });
+      const result = interaction.request();
+      expect(result).toEqual(singleInputB64PSBT);
+    });
+    // TODO: create the PSBT from other parameters instead
+    it("return null if there is no psbt", () => {
+      const interaction = interactionBuilder({});
+      const result = interaction.request();
+      expect(result).toBeNull();
+    });
+  });
+
+  describe("parse", () => {
+    it("return single input, single signature set", () => {
+      const interaction = interactionBuilder({});
+      const result = interaction.parse(singleInputB64PSBT_partiallySigned);
+      const expectedOutput = {
+        '029e866b92b02c686f89b320c0ce4c2391ee531b4dccfd719aefc8caa82e42c828':
+          ['3045022100fd9862cc751bd99f4b77b7d80f86701ba67e667912b2497f83f0d67d95c976ed02205ae2c4f01a361498709a5605baee70b985a9882c55ad022a8fed71e588f1350601'],
+      };
+      expect(result).toEqual(expectedOutput);
+      expect(Object.keys(result).length).toEqual(1);
+    });
+    it("return single input, double signature set", () => {
+      const interaction = interactionBuilder({});
+      const result = interaction.parse(singleInputB64PSBT_fullySigned);
+      const expectedOutput = {
+        '029e866b92b02c686f89b320c0ce4c2391ee531b4dccfd719aefc8caa82e42c828':
+          ['3045022100fd9862cc751bd99f4b77b7d80f86701ba67e667912b2497f83f0d67d95c976ed02205ae2c4f01a361498709a5605baee70b985a9882c55ad022a8fed71e588f1350601'],
+        '02315cbe4eada106463e04c238ffa3dbf26a77ad5047ac769c43f4629525266f45':
+          ['30450221008d633eccf112996792f02d5f39eef951ac042b831517fa3d245ba9b31635d93302207a36defbe2ee3f5d771fce61a801c77e4a00a4ef9a7738c510462339436a601401'],
+      };
+      expect(result).toEqual(expectedOutput);
+      expect(Object.keys(result).length).toEqual(2);
+    });
+    it("return multi input, single signature set", () => {
+      const interaction = interactionBuilder({});
+      const result = interaction.parse(multiInputB64PSBT_partiallySigned);
+      const expectedOutput = {
+        '029e866b92b02c686f89b320c0ce4c2391ee531b4dccfd719aefc8caa82e42c828':
+          [
+            '3045022100bc321b8f03605800e6ccefeba62e8c95e897fdfa2611b2531528b45c5c8c3e9a02200e4497e3abbbbe7aee0dca97363801fc306f22d0aa6c7f81775a11fb1f81598201',
+            '3045022100a41cd24ac399c24449334d0a46906626de003e2b88a9c37e46f2cb0cbfbac0f602203c7524ec1c8f12d535a2212632865da1bef43bbe81c1b20ead12b6cbf7f44f9a01',
+          ],
+      };
+      expect(result).toEqual(expectedOutput);
+      expect(Object.keys(result).length).toEqual(1);
+    });
+    it("return multi input, double signature set", () => {
+      const interaction = interactionBuilder({});
+      const result = interaction.parse(multiInputB64PSBT_fullySigned);
+      const expectedOutput = {
```

Comment:
I also feel like these would make sense in the fixtures file next to the strings that they are expected to be extracted from, but I could be misunderstanding these pieces. It would make the most sense if the outputs are the same regardless of what's doing the interaction, which it feels like should be the case here, but I could be wrong about that.